### DISCUSSION TITLE
Issue with rotation.byAxisAngle

### DISCUSSION THREAD

**Discussion by Ronit-Roy** [mtex-toolbox/mtex#2161] [Ask Anything] [answered]
Hi everyone,

I'm encountering an issue while attempting to rotate my EBSD data along the x-axis using rotation.byAxisAngle. Every time I try, I receive the following error: "_The class rotation has no Constant property or Static method named 'byAxisAngle'_". I've followed various suggestions from the forum, but none have resolved the problem.

Could this issue be related to the versions of MTEX (5.1.1) or MATLAB (2019b) that I'm using? Any advice or suggestions would be greatly appreciated.

Many thanks,
Ronit

**Comment by Tijmenvermeij** [comment]
Can you post the code you use? That might help.

**Reply by Ronit-Roy** [reply]
Hi Tijmen, Thanks for your response.

I was using this:

```
% Analysis
% clean up all non-Zr Alpha data points
ebsd=ebsd('indexed');

% Cut EBSD data
load('MapSize.mat')
ind = inpolygon(ebsd,MapSize);
ebsd = ebsd(ind);

% Rotate data
ebsd = rotate(ebsd,179*degree)   % Rotation along z-axis
ebsd = rotation(ebsd,rotation.byAxisAngle(xvector,-5*degree)) % Rotation along x-axis
```

Many thanks,
Ronit

**Comment by ralfHielscher** [comment, marked as answer]
Yes, this is due to the MTEX version. The syntax `rotation.byAxisAngle` was introduces in MTEX 5.2. The MTEX 5.1 syntax was
``` matlab
R = rotation('axis',v,'angle',omega)
```
Ralf.

**Reply by Ronit-Roy** [reply]
Hi Ralf, 

Thanks for clarifying. I will update my code accordingly.

Many thanks,
Ronit
